test/test-udp-mmsg.c

```diff
@@ -82,12 +83,18 @@ static void recv_cb(uv_udp_t* handle,
     return;
   }
 
-  ASSERT_EQ(nread, 4);
-  ASSERT_NOT_NULL(addr);
-  ASSERT_MEM_EQ("PING", rcvbuf->base, nread);
+  if (nread == 0) {
+    /* There can be no more available data for the time being. */
```

Comment:
There will never be more. The kernel just reported the recv buffer is empty.

Reply:
@vtjnash that's not what I'm observing in my 10.15 OSX. I see that it may take up to several loop iterations to read all the datagrams when running the test with the [2nd modification you proposed.](https://github.com/libuv/libuv/issues/3479#issuecomment-1040677181)